File: examples/quickstart/runtime-embed.go
package main

import (
	"log"
	"time"

	"github.com/axisrobo/harmovela/runtime"
)

func main() {
	config := runtime.DefaultConfig()
	config.Delivery.Store = "memory"
	config.Transports.WebSocket.Enabled = false
	config.Transports.SSE.Enabled = false
	config.Transports.API.Enabled = false

	svc := runtime.NewRuntimeService(config)
	svc.Subscribe("task.*", func(event map[string]any) {
		log.Printf("received %v %v", event["type"], event["id"])
	})

	svc.Start()
	defer svc.Stop()

	svc.Publish(map[string]any{
		"spec_version": "0.2",
		"id":          "evt_embed",
		"type":        "task.submitted",
		"source":      "example:quickstart",
		"created_at":  time.Now().UTC().Format(time.RFC3339),
		"payload":     map[string]any{"task_id": "task_01"},
	})
}

File: examples/quickstart/runtime-embed.java
package com.axisrobo.harmovela.examples;

import com.axisrobo.harmovela.runtime.HarmovelaRuntimeService;
import com.axisrobo.harmovela.runtime.Config;

public class Quickstart {
    public static void main(String[] args) {
        var config = Config.defaultConfig().withStore("memory")
            .withWebsocketEnabled(false).withSseEnabled(false)
            .withApi(new Config.Transport(false, "127.0.0.1", 0, "/harmovela/api"));

        var svc = new HarmovelaRuntimeService(config);
        svc.subscribe("task.*", e ->
            System.out.println("received " + e.get("type") + " " + e.get("id")));

        svc.start();
        svc.publish(java.util.Map.of(
            "spec_version", "0.2",
            "id", "evt_embed",
            "type", "task.submitted",
            "source", "example:quickstart",
            "created_at", java.time.Instant.now().toString(),
            "payload", java.util.Map.of("task_id", "task_01")
        ));
        svc.stop();
    }
}
